products/BellHybrid/apps/Application.cpp

```diff
@@ -134,7 +142,28 @@ namespace app
     sys::MessagePointer Application::handleKBDKeyEvent(sys::Message *msgl)
     {
         onKeyPressed();
-        return ApplicationCommon::handleKBDKeyEvent(msgl);
+        if (this->getState() != app::ApplicationCommon::State::ACTIVE_FORGROUND) {
+            LOG_FATAL("Terrible terrible damage! Application with no focus grabbed key!");
+        }
+        const auto msg        = static_cast<sevm::KbdMessage *>(msgl);
+        const auto inputEvent = keyTranslator->translate(msg->key);
+        const auto key        = mapKey(inputEvent.getKeyCode());
+        if (!inputEvent.is(gui::KeyCode::KEY_UNDEFINED) && inputEvent.isShortRelease() && key != KeyMap::Frontlight &&
+            key != KeyMap::DeepPressDown && key != KeyMap::DeepPressUp && alarmModel->isPreWakeUpActive()) {
```

Comment:
If I see correctly, the synchro message is sent here? IMO it shouldn't be like that :/ please don't merge this, I want to take a closer look at it